content/kb/tutorials/databases/snowflake.md

```diff
@@ -107,39 +104,39 @@ Copy the code below to your Streamlit app and run it. Make sure to adapt the que
 import streamlit as st
 
 # Initialize connection.
-conn = st.experimental_connection('snowpark')
+conn = st.connection("snowflake", type="snowflake")
 
 # Perform query.
-df = conn.query('SELECT * from mytable;', ttl=600)
+df = conn.query("SELECT * from mytable;", ttl=600)
 
 # Print results.
 for row in df.itertuples():
     st.write(f"{row.NAME} has a :{row.PET}:")
 ```
 
-See `st.experimental_connection` above? This handles secrets retrieval, setup, query caching and retries. By default, `query()` results are cached without expiring. In this case, we set `ttl=600` to ensure the query result is cached for no longer than 10 minutes. You can also set `ttl=0` to disable caching. Learn more in [Caching](/library/advanced-features/caching).
+See `st.connection` above? This handles secrets retrieval, setup, query caching and retries. By default, `query()` results are cached without expiring. In this case, we set `ttl=600` to ensure the query result is cached for no longer than 10 minutes. You can also set `ttl=0` to disable caching. Learn more in [Caching](/library/advanced-features/caching).
 
 If everything worked out (and you used the example table we created above), your app should look like this:
 
 ![Finished app screenshot](/images/databases/snowflake-app.png)
 
 ### Using a Snowpark Session
 
-The same [SnowparkConnection](/library/api-reference/connections/st.connections.snowparkconnection) used above also provides access to the [Snowpark Session](https://docs.snowflake.com/en/developer-guide/snowpark/reference/python/session.html) for DataFrame-style operations that run natively inside Snowflake. Using this approach, you can rewrite the app above as follows:
+The same [SnowflakeConnection](/library/api-reference/connections/st.connections.snowflakeconnection) used above also provides access to the [Snowpark Session](https://docs.snowflake.com/en/developer-guide/snowpark/reference/python/session.html) for DataFrame-style operations that run natively inside Snowflake. Using this approach, you can rewrite the app above as follows:
```

Comment:
Developer needs to do `pip install snowflake-snowpark-python` to get `conn.session()` to work. Suggest adding a note or a sentence about that.

Reply:
Done in [6f5c73b](https://github.com/streamlit/docs/pull/860/commits/6f5c73b9affdee566e3a4d81f05cac6992deb3e5)